hack/check-pr.sh

```diff
@@ -56,16 +40,47 @@ step() {
     return 0
 }
 
-failing=()
-step hack/verify-bazel.sh hack/verify-bazel.sh || failing+=("bazel")
+quit() {
+    if [[ "${#failing[@]}" != 0 ]]; then
+        echo "FAILURE: ${#failing[@]} steps failed: ${failing[@]}"
+        exit 1
+    fi
+    echo "SUCCESS"
+    exit 0
+}
+
+step hack/verify-bazel.sh hack/verify-bazel.sh || failing+=("verify-bazel")
 step hack/verify-gofmt.sh hack/verify-gofmt.sh || failing+=("gofmt")
-step //:golint bazel run //:golint -- "${dirs[@]}" || failing+=("golint")
-step //:govet bazel run //:govet -- "${dirs[@]}" || failing+=("govet")
-step "bazel test" bazel test --build_tests_only "${tests[@]}" || failing+=("bazel test")
 step hack/verify_boilerplate.py hack/verify_boilerplate.py || failing+=("boilerplate")
```

Comment:
Aka move this line after verify-bazel.sh and then:
* only run golint and govet if -n dirs
* only run bazel test if -n tests

The rest seems like unnecessary churn